lib/FilterHandler.js

```diff
@@ -2,22 +2,23 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import { filterConfig } from './proptypes';
 import TextFilter from './TextFilter';
+import warning from 'warning';
 
-
-const defaultFilterFunction = ({displayValue, expression}) => {
+const defaultFilterFunction = ({displayValue, expression, columnName}) => {
     if (!(typeof expression === 'string') || expression === ''){
         return true;
     }
     if (displayValue === undefined || displayValue === null){
         return false;
     }
     const valueType = typeof displayValue;
-    if ((valueType === 'string' ) && displayValue.toLowerCase().indexOf(expression.toLowerCase()) === -1){
-        return false;
-    } else if (['number', 'boolean', 'object'].indexOf(valueType) !== -1){
-        return JSON.stringify(displayValue).toLowerCase().indexOf(expression.toLowerCase()) !== -1;
+    if (valueType === 'string'){
+        return displayValue.toLowerCase().indexOf(expression.toLowerCase()) !== -1;
+    }
+    if (process.env.NODE_ENV !== 'production'){
```

Comment:
@fkretzer, do you think that there is any harm if we support `number` too?

Reply:
No, i think number can be supported as well.